app/src/main/java/com/p2p/framework/bluetooth/BluetoothConnectionThread.kt

```diff
@@ -24,45 +24,58 @@ class BluetoothConnectionThread(
 
     private val inputStream: InputStream = socket.inputStream
     private val outputStream: OutputStream = socket.outputStream
-    private val buffer: ByteArray = ByteArray(1024) // mmBuffer store for the stream
+    private val buffer: ByteArray = ByteArray(BUFFER_SIZE)
 
     init {
         start()
     }
 
     override fun run() {
-        var numBytes: Int // bytes returned from read()
-
         // Keep listening to the InputStream until an exception occurs.
-        while (true) {
-            // Read from the InputStream.
-            numBytes = try {
-                // This method blocks until input data is available, end of file is detected, or an exception is thrown.
-                Logger.d(TAG, "Reading")
-                inputStream.read(buffer)
-            } catch (e: IOException) {
-                Logger.d(TAG, "Input stream was disconnected", e)
-                onConnectionLost?.invoke()
-                break
+        infinityLoop@ while (true) {
+            var numBytes = readFromStream() ?: break
+            val isConversation = buffer[IS_CONVERSATION_INDEX].toInt().toBoolean()
+            val packages = buffer[PACKAGES_COUNT_INDEX].toInt()
+            var byteArray = buffer.copyOfRange(EXTRA_INFO_SIZE, numBytes)
+            Logger.d(TAG, "Read 1/$packages message...")
+            for (i in 1 until packages) {
```

Comment:
Que significa el "in 1", no es lo que esta loopeando? *confused

Reply:
Es la forma de kotlin de hacer un for de 1 a N 😄